### PR TITLE
Add 'valid_locale-pattern' configuration key

### PR DESCRIPTION
**Pull request by jmleroux**
When cleaning translation files, Nelson compare the Crowdin file with a `locale_map` array.

If the locale is not in this array, Nelson will then transform the locale name to extract a '_simple locale_' which can be on only two letters. The two letters is not acceptable anymore for PIM >= 3.2, so I added a configuration key with a regex pattern to check the resulting simple locale. If the configuration key is not set, it is defaulting to `/^.+$/`, ie basically anything with at least one character.

This way, Nelson stays compatible with older version and we will have more freedom to validate the locale pattern.

Any file not in the locale map or complying to the pattern will not be renamed and not be pushed to the PIM. We will have to add it to the locale map if we want to integrate it.

### CODE REVIEW

File: src/Akeneo/Git/ProjectCloner.php [hunk 1]
```diff
@@ -223,11 +223,11 @@ protected function update($baseBranch, $projectDir)
...
-        $this->executor->execute(sprintf(
-            'cd %s && git push origin %s',
-            $projectDir,
-            $baseBranch
-        ));
+//        $this->executor->execute(sprintf(
+//            'cd %s && git push origin %s',
+//            $projectDir,
+//            $baseBranch
+//        ));
```
Comment (pierallard) [inline src/Akeneo/Git/ProjectCloner.php:230]: Are you sure about this ?

File: config/example.yml [hunk 2]
```diff
@@ -30,6 +30,8 @@ crowdin:
         locale_map:
             fr_FR: fr
             fr_BE: fr_BE
+        # You can specify a valid locale pattern to check after before renaming the files
+        valid_locale_pattern: /^[a-z]{2}_[A-Z]{2}$/
```
Comment (pierallard) [inline config/example.yml:34]: Warning, I think there is some strange locales with 3 or 4 locale parameter

gsw-FR
kok-IN
quz-EC
arn-CL

**Review by pierallard — APPROVED** [review]

File: src/Akeneo/Git/ProjectCloner.php (hunk 1 above)
Comment (jmleroux) [reply src/Akeneo/Git/ProjectCloner.php]: whoopsy commit

File: config/example.yml (hunk 2 above)
Comment (jmleroux) [reply config/example.yml:34]: Ah...
I'll have to set the right pattern then. Thanks for the hint!

**Pull request merged**